src/stream/src/common/table/state_table.rs

```diff
@@ -471,8 +485,9 @@ impl<S: StateStore> StateTable<S> {
                     .into_iter()
                     .map(|index| self.pk_indices[index])
                     .collect_vec();
+
                 let read_options = ReadOptions {
-                    prefix_hint: None,
+                    dist_key_hint: None,
                     check_bloom_filter: self.dist_key_indices == key_indices,
```

Comment:
After discussion, we decided to check bloom filter when `input_pk==dist_key`

Reply:
FYI, we should also check `input_pk` is not empty, because if `dist_key` and `input_pk` is both empty, we can not check bloom filter.